Support external LINSTOR Controllers (#451)
piraeusdatastore/piraeus-operator: docs/how-to/external-controller.md

# How to Use an Existing LINSTOR Cluster

This guide shows you how to connect Piraeus with an existing LINSTOR® Cluster, managed outside Kubernetes.

To complete this guide, you should be familiar with:

* editing `LinstorCluster` resources.

## Configure the Piraeus Resources

To use an externally managed LINSTOR Cluster, specify the URL of the LINSTOR Controller in the `LinstorCluster`
resource. In the following example, the Controller is reachable at `http://linstor-controller.example.com:3370`:

```yaml
apiVersion: piraeus.io/v1
kind: LinstorCluster
metadata:
  name: linstorcluster
spec:
  externalController:
    url: http://linstor-controller.example.com:3370
```

After applying the `LinstorCluster`, Piraeus will make the following changes to the default deployment:

* The `linstor-controller` Deployment is removed
* The `linstor-csi-controller` and `linstor-csi-node` resources will reference the external controller.

## Configuring Host Networking for LINSTOR Satellites

> Next, configure your Satellites to use [host networking](./drbd-host-networking.md). Unless your Pod network is
> reachable from outside the Kubernetes Cluster, the existing LINSTOR Cluster won't be able to communicate with the
> Satellites in the Kubernetes cluster.

Reviewer: Slightly simpler logical flow IMHO:

```suggestion
Normally the Pod network is not reachable from outside the Kubernetes Cluster.
In this case the existing LINSTOR Controller won't be able to communicate with the Satellites in the Kubernetes cluster.
So you should configure your Satellites to use [host networking](./drbd-host-networking.md).
```

Author: Thanks, applied!
